nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -2254,6 +2242,123 @@ mod tests {
+        // While this is *mostly* made-up blueprint contents, the part that matters
+        // is that it's provisioning a zone (Nexus) which does have resources
+        // to be allocated.
+        let ip_range = IpRange::try_from((
+            Ipv4Addr::new(10, 0, 0, 1),
+            Ipv4Addr::new(10, 0, 0, 10),
+        ))
+        .unwrap();
+        let (service_ip_pool, _) = datastore
+            .ip_pools_service_lookup(&opctx)
+            .await
+            .expect("lookup service ip pool");
+        datastore
+            .ip_pool_add_range(&opctx, &service_ip_pool, &ip_range)
+            .await
+            .expect("add range to service ip pool");
+        let zone_id = OmicronZoneUuid::new_v4();
+        blueprint.blueprint_zones.insert(
+            sled_id,
+            BlueprintZonesConfig {
+                generation: omicron_common::api::external::Generation::new(),
+                zones: vec![BlueprintZoneConfig {
+                    disposition: BlueprintZoneDisposition::InService,
+                    id: zone_id,
+                    underlay_address: Ipv6Addr::LOCALHOST,
+                    filesystem_pool: None,
+                    zone_type: BlueprintZoneType::Nexus(
+                        blueprint_zone_type::Nexus {
+                            internal_address: SocketAddrV6::new(
+                                Ipv6Addr::LOCALHOST,
+                                0,
+                                0,
+                                0,
+                            ),
+                            external_ip: OmicronZoneExternalFloatingIp {
+                                id: ExternalIpUuid::new_v4(),
+                                ip: "10.0.0.1".parse().unwrap(),
+                            },
+                            nic: NetworkInterface {
+                                id: Uuid::new_v4(),
+                                kind: NetworkInterfaceKind::Service {
+                                    id: *zone_id.as_untyped_uuid(),
+                                },
+                                name: Name::from_str("mynic").unwrap(),
+                                ip: "fd77:e9d2:9cd9:2::8".parse().unwrap(),
+                                mac: MacAddr::random_system(),
+                                subnet: IpNet::host_net(IpAddr::V6(
+                                    Ipv6Addr::LOCALHOST,
+                                )),
+                                vni: Vni::random(),
+                                primary: true,
+                                slot: 1,
+                                transit_ips: vec![],
+                            },
+                            external_tls: false,
+                            external_dns_servers: vec![],
+                        },
+                    ),
+                }],
+            },
+        );
+
+        blueprint
+    }
+
+    #[tokio::test]
+    async fn test_ensure_external_networking_works_with_good_target() {
```

Comment:
I split this into a new test because it's **really important** that we actually perform a write **after** checking the target in the "bad case" below.

If we don't actually perform a write, then there is no circular dependency between transactions, and CockroachDB can logically re-order the "ensure resources" task before other operations that might write to the target, regardless of when it *actually* happened.